### PR TITLE
packaging/{opensuse,fedora}: allow package build with testkeys included

### PR DESCRIPTION
**Pull request by morphis**
Special build with test keys enabled is necessary for a few spread tests.

### CODE REVIEW

File: packaging/fedora/snapd.spec [hunk 1]
```diff
@@ -320,11 +331,22 @@ export GOPATH=$(pwd):%{gopath}
 export GOPATH=$(pwd):$(pwd)/Godeps/_workspace:%{gopath}
 %endif
 
-%gobuild -o bin/snap %{import_path}/cmd/snap
-%gobuild -o bin/snap-exec %{import_path}/cmd/snap-exec
-%gobuild -o bin/snapctl %{import_path}/cmd/snapctl
-%gobuild -o bin/snapd %{import_path}/cmd/snapd
-%gobuild -o bin/snap-update-ns %{import_path}/cmd/snap-update-ns
+# We have to build snapd first to prevent the build from
```
Comment (zyga) [inline packaging/fedora/snapd.spec]: The comment seems to be copy pasted, is this intended?

File: packaging/fedora/snapd.spec [hunk 2]
```diff
@@ -320,11 +331,22 @@ export GOPATH=$(pwd):%{gopath}
...
+# building various things from the tree without additional
+# set tags.
+GOFLAGS=
+%if 0%{?with_test_keys}
+GOFLAGS="-tags withtestkeys"
+%endif
+
+# We have to build snapd first to prevent the build from
```
Comment (zyga) [inline packaging/fedora/snapd.spec]: I don't think anything else cares about test keys, if that lets you simplify things, go for it!

File: packaging/opensuse-42.2/snapd.spec [hunk 3]
```diff
@@ -117,11 +127,22 @@ export CXXFLAGS
 %build
 # Build golang executables
 %goprep %{import_path}
+
+%if 0%{?with_test_keys}
+# The %gobuild macro doesn't allow us to pass any additional parameters
+# so we we have to invoke `go install` here manually.
+export GOPATH=%{_builddir}/go:%{_libdir}/go/contrib
+export GOBIN=%{_builddir}/go/bin
+go install -s -v -p 4 -x -tags withtestkeys github.com/snapcore/snapd/cmd/snapd
```
Comment (zyga) [inline packaging/opensuse-42.2/snapd.spec]: Nitpick, can you use long forms or document the meaning of the switches above in a comment, for the next reader?

Comment (morphis) [inline packaging/opensuse-42.2/snapd.spec]: You mean for `-x -v -p 4 -x`

Comment (morphis) [inline packaging/opensuse-42.2/snapd.spec]: Done.

File: packaging/fedora/snapd.spec [hunk 4]
```diff
@@ -320,11 +331,22 @@ export GOPATH=$(pwd):%{gopath}
...
-%gobuild -o bin/snapctl %{import_path}/cmd/snapctl
-%gobuild -o bin/snapd %{import_path}/cmd/snapd
-%gobuild -o bin/snap-update-ns %{import_path}/cmd/snap-update-ns
+# We have to build snapd first to prevent the build from
+# building various things from the tree without additional
+# set tags.
+GOFLAGS=
+%if 0%{?with_test_keys}
+GOFLAGS="-tags withtestkeys"
+%endif
```
Comment (morphis) [inline packaging/fedora/snapd.spec]: Fixed

File: packaging/fedora/snapd.spec (hunk 1 above)
Comment (morphis) [inline packaging/fedora/snapd.spec]: Removed one of both.

File: packaging/fedora/snapd.spec [hunk 5]
```diff
@@ -320,11 +331,19 @@ export GOPATH=$(pwd):%{gopath}
 export GOPATH=$(pwd):$(pwd)/Godeps/_workspace:%{gopath}
 %endif
 
-%gobuild -o bin/snap %{import_path}/cmd/snap
-%gobuild -o bin/snap-exec %{import_path}/cmd/snap-exec
-%gobuild -o bin/snapctl %{import_path}/cmd/snapctl
-%gobuild -o bin/snapd %{import_path}/cmd/snapd
-%gobuild -o bin/snap-update-ns %{import_path}/cmd/snap-update-ns
+GOFLAGS=
```
Comment (Conan-Kudo) [inline packaging/fedora/snapd.spec]: You're still erasing the GOFLAGS here... :(

Comment (morphis) [inline packaging/fedora/snapd.spec]: Who else uses GOFLAGS? Nothing in the spec file does and nothing of the RPM golang macros does.

**Pull request merged**